specification/recoveryservices/Recoveryservices.Management/models.tsp

```diff
@@ -0,0 +1,1685 @@
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Standard: "Standard",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  RS0: "RS0",
+}
+
+/**
+ * Unit of the usage.
+ */
+union UsagesUnit {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Count: "Count",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Bytes: "Bytes",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Seconds: "Seconds",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Percent: "Percent",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  CountPerSecond: "CountPerSecond",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  BytesPerSecond: "BytesPerSecond",
+}
+
+/**
+ * Details of the certificate to be uploaded to the vault.
+ */
+model CertificateRequest {
+  /**
+   * Raw certificate data.
+   */
+  properties?: RawCertificateData;
+}
+
+/**
+ * Raw certificate data.
+ */
+model RawCertificateData {
+  /**
+   * Specifies the authentication type.
+   */
+  authType?: AuthType;
+
+  /**
+   * The base64 encoded certificate raw data string
+   */
+  certificate?: bytes;
+}
+
+/**
+ * Certificate corresponding to a vault that can be used by clients to register themselves with the vault.
+ */
+model VaultCertificateResponse {
+  /**
+   * Resource name associated with the resource.
+   */
+  @visibility(Lifecycle.Read)
+  name?: string;
+
+  /**
+   * Resource type represents the complete path of the form Namespace/ResourceType/ResourceType/...
+   */
+  @visibility(Lifecycle.Read)
+  type?: string;
+
+  /**
+   * Resource Id represents the complete path to the resource.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+
+  /**
+   * Certificate details representing the Vault credentials.
+   */
+  properties?: ResourceCertificateDetails;
+}
+
+/**
+ * Certificate details representing the Vault credentials.
+ */
+@discriminator("authType")
+model ResourceCertificateDetails {
+  /**
+   * This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  authType: string;
```

Comment:
>string [](http://example.com/codeflow?start=12&length=6)

what is no string discriminator complaining about - that it should be an enum??

Reply:
This would look like a breaking change to fix within this version right. Do we file a ticket for service team to fix it in their next api version?